src/uwtools/config/formats/base.py

```diff
@@ -165,7 +171,7 @@ def dereference(self, context: Optional[dict] = None) -> None:
 
         def logstate(state: str) -> None:
             log.debug("Dereferencing, %s value:", state)
-            for line in str(self).split("\n"):
+            for line in yaml_to_str(self.data).split("\n"):
```

Comment:
Previously, we could use `str()` here because all `Config` subclasses inherited a `__repr__()` that represented their `.data` `dict`s as YAML. Now, `str()` will return the format-appropriate string, but for debugging the operation of the `dereference()` operation, YAML remains a convenient and consistent way to represent the `.data` `dict` as it is repeatedly dereferenced until no more Jinja2 expressions remain.